main.php

```diff
@@ -47,13 +47,19 @@ function ($errno, $errstr, $errfile, $errline, array $errcontext) {
 
     // ZIP
     $finder = new \Symfony\Component\Finder\Finder();
-    $finder->name("*.zip")->in($dataFolder . "/in/files")->depth(0);
+    $finder->name("*.zip")->in($dataFolder . "/in/files");
     foreach ($finder as $sourceFile) {
         try {
+            $subfolder = substr($sourceFile->getPath(), strlen($dataFolder . 'in/files/'));
+            if ($subfolder) {
+                if (!$fs->exists($dataFolder . "/out/files" . $subfolder)) {
+                    $fs->mkdir($dataFolder . "/out/files" . $subfolder);
+                }
+            }
             (new \Symfony\Component\Process\Process(
-                "unzip -j {$sourceFile->getPathname()} -d {$dataFolder}/out/files/{$sourceFile->getBasename()}"
-            )
-            )->mustRun();
+                "unzip {$sourceFile->getPathname()} -d {$dataFolder}/out/files{$subfolder}/{$sourceFile->getBasename()}"
+            ))
```

Comment:
kdyz se vyhodi to `/{$sourceFile->getBasename()}` tak to pak bude podporovat i ty sliced files - toto prece neni k nicemu potreba ne?

Reply:
pokud v zipu není adresář, tak to rozbalí do rootu.

v zipu `archive.zip` jsou dva soubory, `file1`, `file2`

`unzip {$sourceFile->getPathname()} -d {$dataFolder}/out/files{$subfolder}/{$sourceFile->getBasename()}`
=>
```
/data/out/files/archive.zip/file1
/data/out/files/archive.zip/file2
```

`unzip {$sourceFile->getPathname()} -d {$dataFolder}/out/files{$subfolder}/`
=>
```
/data/out/files/file1
/data/out/files/file2
```
(proměnná `$subfolder` je prázdná)

Reply:
no vsak ano - gzip to taky rozbali to 'rootu'

Reply:
aha, no tak možná jo teda :-) chtěl jsem ze zipů rovnou udělat slicovaný soubory, ale to by asi teda mohl udělat nějakej další procesor

Reply:
jak, ze zipu rovnou slicovany soubory? to prece v gz nemas taky ne? proste mi jde akorat o to, aby se to chovalo stejne pro zip a pro gz

Reply:
dobrá! z nějakýho důvodu jsem gzipy rozbaloval tam kde byly a zipy do podadresáře (use case -je jen jeden archiv, u gzipu po rozbalení zůstane, u zipu může bejt souborů víc)

Reply:
jo a mimochodem adresářová struktura uvnitř zipu je zachovaná